ansible_cloudstack_utils.py

```diff
@@ -219,7 +219,13 @@ def get_zone(self, key=None):
 
         # use the first zone if no zone param given
         if zones:
-            self.zone = zones['zone'][0]
+            # Some old cloudstack has bug which does not recognize listZones?name=xxx parameter.
+            # Filter manually.
+            tmp = filter(lambda x: x['name'] == zone, zones['zone'])
```

Comment:
I actually changed this recently, it was the way you fixed it before. So this looks good to me expect that you must handle the zone name in a case insensitive way: `x['name'].lower() == zone.lower()`